### PR TITLE
Refactor checkout tax interface in plugins 

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because of refactoring the checkout tax interface in plugins.

1. Drop the concept of `undiscounted_price`, `price_with_sale`, and `price_with_discounts` instead of that taxes should be calculated on the final price. 
2. Refactor SyncTax input interface to be consistent with description from #9526 

:warning: 
This PR is to feature branch not to main. 
Migrations will be refacotred in #9526.

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migration files are up to date
* [ ] The changes are tested
* [ ] GraphQL schema and type definitions are up to date
* [ ] Changes are mentioned in the changelog

### CODE REVIEW

File: saleor/checkout/base_calculations.py [hunk 1]
```diff
@@ -26,19 +25,8 @@ def calculate_base_line_unit_price(
...
-    Voucher is added to 'price_with_discounts' when line's product matches to products
-    applicable for voucher.
-    For voucher with 'apply once per order', the price will be reduced in calculation
-    of total price.
-    'price' includes discount from sale if any valid exists.
-    'price_with_discounts' includes voucher discount and sale discount if any valid
-    exists.
-    'undiscounted_price' is a price without any sale and voucher.
-    """
+) -> Money:
```
Comment (korycins) [inline saleor/checkout/base_calculations.py:28]: Would be good to have some description here. Especially that we're applying here voucher

Comment (korycins) [reply saleor/checkout/base_calculations.py:28]: Somethin like:
```
"""Calculate line unit price.
Unit price includes discount from sale and voucher
A voucher is added to the unit price when line's product matches to products applicable for the voucher.
"""
```

File: saleor/webhook/serializers.py [hunk 2]
```diff
@@ -87,7 +87,9 @@ def untaxed_price_amount(price: TaxedMoney) -> Decimal:
     return [
         {
             **_get_checkout_line_payload_data(line_info),
-            "unit_amount": untaxed_price_amount(line_info.line.unit_price),
+            "unit_amount": untaxed_price_amount(
+                line_info.line.total_price / line_info.line.quantity
```
Comment (korycins) [inline saleor/webhook/serializers.py:91]: Not sure if `untaxed_price_amount` already does this, but keep in mind that you should quantize_price here

Comment (fowczarek) [reply saleor/webhook/serializers.py:91]: image
`quantize_price` is used inside `untaxed_price_amount`.
Do you want to extract it from this function?

Comment (korycins) [reply saleor/webhook/serializers.py:91]: IMO we can extract it

**Review by IKarbowiak — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]
We should remember about updating Avatax Excise plugin.

**Pull request merged**